src/Functions/FunctionFile.cpp

```diff
@@ -30,21 +34,39 @@ class FunctionFile : public IFunction, WithContext
     static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionFile>(context_); }
     explicit FunctionFile(ContextPtr context_) : WithContext(context_) {}
 
+    bool isVariadic() const override { return true; }
     String getName() const override { return name; }
-    size_t getNumberOfArguments() const override { return 1; }
+    size_t getNumberOfArguments() const override { return 0; }
     bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
 
     DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
     {
+        if (arguments.empty() || arguments.size() > 2)
+            throw Exception(
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Number of arguments for function {} doesn't match: passed {}, should be 1 or 2",
+                getName(), toString(arguments.size()));
+
         if (!isString(arguments[0].type))
             throw Exception(ErrorCodes::NOT_IMPLEMENTED, "{} is only implemented for type String", getName());
 
+        if (arguments.size() == 2)
+        {
+            if (arguments[1].type->onlyNull())
+                return makeNullable(std::make_shared<DataTypeString>());
+
+            if (!isString(arguments[1].type))
+                throw Exception(ErrorCodes::NOT_IMPLEMENTED, "{} only accepts String or Null as second argument", getName());
+        }
+
         return std::make_shared<DataTypeString>();
     }
 
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
```

Comment:
```suggestion
    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
    
    bool useDefaultImplementationForNulls() const override { return false; }
```
We should also disable default implementation for Nulls, otherwise if second argument is `NULL`, result will be always `NULL` with type `Nullable(Nothing)` even when file exists.